unitTests/ui/CoreHelpersTest.ts

```diff
@@ -95,6 +96,14 @@ export function CoreHelperTest() {
       );
     });
 
+    it('highlightStreamText should not crash in case of duplicate terms on a text that starts with a word delimiter', () => {
+      const toHighlight = `'This is a bad text`;
+      const terms: IHighlightTerm = { aTerm: [] };
+      const phrases: IHighlightPhrase = { aTerm: terms };
+
+      expect(TemplateHelpers.getHelper('highlightStreamText')(toHighlight, terms, phrases)).toEqual(_.escape(toHighlight));
```

Comment:
Why is `toHighlight` escaped in the assertion?

Reply:
The highlightStreamText helper will escape the text to highlight; the single quote will get html encoded as part of its normal operation.

Reply:
Since the test description is `should not crash`, would `expect(()=> TemplateHelpers.getHelper('highlightStreamText')(toHighlight, terms, phrases)).not.toThrow()` be a better test ?

Reply:
"not crash" might not be the right words, it will "hang", loop/freeze the tests, and crash when there's no more RAM to consume. I tried setting a timeout but the chromium process doesn't stop and keeps looping (I had to task kill it).

I'll rename the test to better describe the issue.